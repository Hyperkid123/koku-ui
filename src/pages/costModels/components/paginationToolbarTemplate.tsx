import {
  Pagination,
  PaginationProps,
  Toolbar,
  ToolbarContent,
  ToolbarItem,
  ToolbarProps,
} from '@patternfly/react-core';
import { TFunction } from 'i18next';
import React from 'react';
import { Translation } from 'react-i18next';

type UniqueProps = Pick<ToolbarProps, 'id' | 'aria-label'>;
type PaginationToolbarTemplateProps = PaginationProps & UniqueProps;

const translatePaginationToolbarProps = (t: TFunction, props: PaginationToolbarTemplateProps) => {
  return {
    ...props,
    'aria-label': t(props['aria-label']),
  };
};

export const PaginationToolbarTemplate: React.FunctionComponent<PaginationToolbarTemplateProps> = props => {
  return (
    <Translation>
      {t => {
        const translatedProps = translatePaginationToolbarProps(t, props);
        const {
          id,
          'aria-label': ariaLabel,
          itemCount,
          perPage,
          page,
          variant,
          onPerPageSelect,
          onSetPage,
        } = translatedProps;
        return (
          <Toolbar id={id}>
            <ToolbarContent aria-label={ariaLabel}>
              <ToolbarItem variant="pagination">
                <Pagination
                  variant={variant}
                  itemCount={itemCount}
                  perPage={perPage}
                  page={page}
                  onSetPage={onSetPage}
                  onPerPageSelect={onPerPageSelect}
                />
              </ToolbarItem>
            </ToolbarContent>
          </Toolbar>
        );
      }}
    </Translation>
  );
};

export default PaginationToolbarTemplate;
