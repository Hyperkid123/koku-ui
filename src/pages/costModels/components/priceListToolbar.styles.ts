import { global_spacer_md } from '@patternfly/react-tokens/dist/esm/global_spacer_md';

export const styles = {
  toolbar: {
    gridGap: global_spacer_md.value,
  },
};
