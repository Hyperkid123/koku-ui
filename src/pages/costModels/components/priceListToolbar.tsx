import {
  PageHeaderTools,
  PageHeaderToolsGroup,
  PageHeaderToolsItem,
  ToolbarContent,
  ToolbarFilter,
  ToolbarItemVariant,
} from '@patternfly/react-core';
import React from 'react';

interface PriceListToolbarProps {
  primary: React.ReactNode;
  selected: string;
  secondaries: {
    component: React.ReactNode;
    name: string;
    onRemove: (category: string, chip: string) => void;
    filters: string[];
  }[];
  onClear: () => void;
  pagination: React.ReactNode;
  button: React.ReactNode;
}

export const PriceListToolbar: React.SFC<PriceListToolbarProps> = ({
  primary,
  secondaries,
  pagination,
  button,
  onClear,
  selected,
}) => {
  return (
    <PageHeaderTools
      clearAllFilters={onClear}
      id="price-list-toolbar"
      style={{ marginBottom: '10px', marginTop: '10px' }}
    >
      <ToolbarContent>
        <PageHeaderToolsGroup variant="filter-group">
          <PageHeaderToolsItem>{primary}</PageHeaderToolsItem>
          {secondaries.map(secondary => {
            return (
              <PageHeaderToolsItem key={secondary.name}>
                <ToolbarFilter
                  deleteChip={secondary.onRemove}
                  chips={secondary.filters}
                  categoryName={secondary.name}
                >
                  {selected === secondary.name ? secondary.component : ''}
                </ToolbarFilter>
              </PageHeaderToolsItem>
            );
          })}
        </PageHeaderToolsGroup>
        <PageHeaderToolsItem>{button}</PageHeaderToolsItem>
        <PageHeaderToolsItem variant={ToolbarItemVariant.pagination}>
          {pagination}
        </PageHeaderToolsItem>
      </ToolbarContent>
      <hr className="pf-c-divider" />
    </PageHeaderTools>
  );
};
