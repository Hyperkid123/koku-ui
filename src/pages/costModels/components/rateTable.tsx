import {
  IActions,
  Table,
  TableBody,
  TableHeader,
  TableVariant,
} from '@patternfly/react-table';
import React from 'react';
import { InjectedTranslateProps } from 'react-i18next';
import { formatCurrency } from 'utils/formatValue';

import { TierData } from './addPriceList';

interface RateTableProps extends InjectedTranslateProps {
  tiers: TierData[];
  actions?: IActions;
  isCompact?: boolean;
}

export const RateTable: React.SFC<RateTableProps> = ({
  t,
  tiers,
  actions,
  isCompact,
}) => {
  return (
    <Table
      aria-label="price-list"
      variant={isCompact ? TableVariant.compact : undefined}
      rows={tiers.map(tier => [
        t(`cost_models.${tier.meta.label_metric}`),
        t(`cost_models.${tier.meta.label_measurement}`, {
          units: tier.meta.label_measurement_unit,
        }),
        `${formatCurrency(Number(tier.rate))}`,
        tier.costType,
      ])}
      cells={[
        t('cost_models.table.metric'),
        t('cost_models.table.measurement'),
        t('cost_models.table.rate'),
        t('cost_models.table.cost_type'),
      ]}
      actions={actions}
    >
      <TableHeader />
      <TableBody />
    </Table>
  );
};
