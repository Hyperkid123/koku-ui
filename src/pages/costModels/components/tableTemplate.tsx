import { Table, TableBody, TableGridBreakpoint, TableHeader, TableProps } from '@patternfly/react-table';
import { TFunction } from 'i18next';
import React from 'react';
import { Translation } from 'react-i18next';

export type TableTemplateProps = Pick<TableProps, 'aria-label' | 'actions' | 'cells' | 'rows' | 'onSort' | 'sortBy'>;

export const translateTableProps = (t: TFunction, props: TableTemplateProps) => {
  const cells = props.cells.map(cell => {
    if (typeof cell === 'string') {
      return t(cell);
    }
    if (typeof cell.title === 'string') {
      return { ...cell, title: t(cell.title) };
    }
    return cell;
  });
  const actions = props.actions.map(action => {
    let act = { ...action };
    if (typeof action.title === 'string') {
      act = { ...act, title: t(action.title) };
    }
    if (action.tooltip && typeof action.tooltip === 'string') {
      act = { ...act, tooltip: t(action.tooltip) };
    }
    return act;
  });
  return {
    ...props,
    cells,
    actions,
  };
};

export const TableTemplate: React.FunctionComponent<TableTemplateProps> = props => {
  return (
    <Translation>
      {t => {
        const translatedProps = translateTableProps(t, props);
        const { actions, cells, rows, onSort, sortBy, 'aria-label': ariaLabel } = translatedProps;
        return (
          <Table
            aria-label={ariaLabel}
            gridBreakPoint={TableGridBreakpoint.grid2xl}
            cells={cells}
            rows={rows}
            sortBy={sortBy}
            onSort={onSort}
            actions={actions}
          >
            <TableHeader />
            <TableBody />
          </Table>
        );
      }}
    </Translation>
  );
};
