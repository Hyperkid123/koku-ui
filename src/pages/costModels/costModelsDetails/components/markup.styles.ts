import { global_FontSize_xl, global_spacer_md } from '@patternfly/react-tokens';
import React from 'react';

export const styles = {
  card: {
    minHeight: '130px',
    maxWidth: '400px',
    marginLeft: global_spacer_md.value,
  },
  cardBody: {
    fontSize: global_FontSize_xl.value,
    textAlign: 'center',
  },
} as { [className: string]: React.CSSProperties };
