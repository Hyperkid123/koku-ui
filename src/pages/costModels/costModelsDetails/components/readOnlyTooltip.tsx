import { Tooltip } from '@patternfly/react-core';
import React from 'react';
import { WithTranslation, withTranslation } from 'react-i18next';

interface ReadOnlyTooltipBase extends WithTranslation {
  children: JSX.Element;
  isDisabled: boolean;
}

export const ReadOnlyTooltipBase: React.SFC<ReadOnlyTooltipBase> = ({ children, t, isDisabled }) => {
  return isDisabled ? (
    <Tooltip isContentLeftAligned content={<div>{t('cost_models.read_only_tooltip')}</div>}>
      <div data-testid="read-only-tooltip">{children}</div>
    </Tooltip>
  ) : (
    children
  );
};

export const ReadOnlyTooltip = withTranslation()(ReadOnlyTooltipBase);
