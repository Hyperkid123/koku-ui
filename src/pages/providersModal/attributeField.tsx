import { FormGroup } from 'components/formGroup';
import { TextInput } from 'components/textInput';
import React from 'react';

export interface AttributeProps {
  label: string;
  autoFocus?: boolean;
  testProps: { [key: string]: string };
  placeholder: string;
  value: string;
  error: string;
}

export interface AttributeChange {
  onChange: (value: string) => void;
}

type Props = AttributeProps & AttributeChange;

const AttributeField: React.SFC<Props> = props => (
  <FormGroup label={props.label}>
    <TextInput
      {...props.testProps}
      autoFocus={Boolean(props.autoFocus)}
      isError={Boolean(props.error)}
      isFlat
      onChange={props.onChange}
      placeholder={props.placeholder}
      value={props.value}
    />
  </FormGroup>
);

export default AttributeField;
