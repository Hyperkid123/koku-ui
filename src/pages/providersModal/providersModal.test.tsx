import { Alert } from '@patternfly/react-core';
import { AxiosError, AxiosResponse } from 'axios';
import { TextInput } from 'components/textInput';
import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';
import { FetchStatus } from 'store/common';
import { Props, ProvidersModal } from './providersModal';

const props: Props = {
  addProvider: jest.fn(),
  clearProviderFailure: jest.fn(),
  closeProvidersModal: jest.fn(),
  fetchStatus: FetchStatus.none,
  isProviderModalOpen: false,
  t: jest.fn(v => v),
};

test('updates provider name on input change', () => {
  const view = shallow(<ProvidersModal {...props} />);
  const input = 'AWSHCCM';
  getNameInput(view).simulate('change', input);
  expect(getNameInput(view).props().value).toBe(input);
});

test('updates bucket name on input change', () => {
  const view = shallow(<ProvidersModal {...props} />);
  const input = 'cost-usage-bucket';
  getBucketInput(view).simulate('change', input);
  expect(getBucketInput(view).props().value).toBe(input);
});

test('updates provider resource name on input change', () => {
  const view = shallow(<ProvidersModal {...props} />);
  const input = 'arn:aws:iam::589173575009:role/CostManagement';
  getResourceNameInput(view).simulate('change', input);
  expect(getResourceNameInput(view).props().value).toBe(input);
});

test('Alert is shown with login error', () => {
  const mockAxiosResponse: AxiosResponse = {
    data: {},
    status: 1,
    statusText: '',
    headers: {},
    config: {},
  };
  const mockAxiosError: AxiosError = {
    config: {},
    message: '',
    name: '',
    response: mockAxiosResponse,
  };
  const view = shallow(<ProvidersModal {...props} error={mockAxiosError} />);
  expect(view.find(Alert)).toMatchSnapshot();
});

function getBucketInput(view: ShallowWrapper<any, any>) {
  return view.find(TextInput).at(0);
}

function getNameInput(view: ShallowWrapper<any, any>) {
  return view.find(TextInput).at(1);
}

function getResourceNameInput(view: ShallowWrapper<any, any>) {
  return view.find(TextInput).at(2);
}
