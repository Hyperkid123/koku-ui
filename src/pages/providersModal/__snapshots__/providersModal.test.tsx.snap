// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert is shown with login error 1`] = `
<Alert
  action={null}
  className=""
  data-testid="alert"
  title="providers.default_error"
  variant="danger"
/>
`;
